### PR TITLE
update to go 1.12

### PR DESCRIPTION
**Pull request by vigith**
Umbrella Issue https://github.com/heptiolabs/eventrouter/issues/82

# Go 1.12
* Use go .12 and use modules instead of dep (mostly from #80)
* remove go mod=vendor
* Updated the versions of:
  * Go: v1.9 -> v1.12.x
  * Alpine Linux: v3.6 -> v3.9.
  * Go dependencies, for example: k8s.io/client_go v5.0.0 -> v12.0.0
* version to 0.3

### CODE REVIEW

**Comment by vigith** [comment]
@johnSchnake Could you please review?

File: Dockerfile [hunk 1]
```diff
@@ -11,21 +11,12 @@
...
-COPY . .
 
-RUN CGO_ENABLED=0 go build -o /src/eventrouter
-
-FROM alpine:3.9 as release
-
-WORKDIR /app
 RUN apk update --no-cache && apk add ca-certificates
-COPY --from=build /src/eventrouter .
+ADD eventrouter /eventrouter
```
Comment (johnSchnake) [inline Dockerfile]: So something is wrong with the image that gets built.

I'm able to build it OK but then it fails to run:
```
$ kubectl logs eventrouter-854dbfdc6d-dt7lv -n kube-system
/bin/sh: /eventrouter: not found
```

When I run the image and look inside it is there:
```
$ pwd
/
$ ls -lah|grep event
-rwxr-xr-x    1 root     root       53.6M Sep 27 20:10 eventrouter
```

I think one issue is that you're putting it into the root directory but we are running as `nobody:nobody`, but even after I manually run the image as a root user I still get the same errors.

I tried moving it into `/app/eventrouter` as well and the same issues. I can see it in `ls`, I can even `cat ./eventrouter|head` and `cp ./eventrouter foo` but I can not run the files. I can `chown` them and `chmod +x ...` them as the root user, but still not run them.

Comment (johnSchnake) [reply Dockerfile]: I found the issue; see comment in makefile.

File: Makefile [hunk 2]
```diff
@@ -31,12 +31,13 @@ TEST = go test $(TEST_PKGS) $(TESTARGS)
...
 VET = go vet $(VET_PKGS)
 
-DOCKER_TEST ?= $(DOCKER) run --rm -v $(DIR):$(BUILDMNT) -w $(BUILDMNT) $(BUILD_IMAGE) /bin/sh -c
+DOCKER_BUILD ?= $(DOCKER) run --rm -v $(DIR):$(BUILDMNT) -w $(BUILDMNT) $(BUILD_IMAGE) /bin/sh -c
 
 all: container
 
 container:
-	$(DOCKER) build -t $(REGISTRY)/$(TARGET):latest --target release -t $(REGISTRY)/$(TARGET):$(VERSION) .
+	$(DOCKER_BUILD) 'go build'
```
Comment (johnSchnake) [inline Makefile]: We need to be building the binary for 32-bit if we are going to be running it on alpine:3.9
```
Linux b789bf130d06 4.9.184-linuxkit #1 SMP Tue Jul 2 22:58:16 UTC 2019 x86_64 Linux
```

I think moving the binary into `/app` is also a good idea so its not in root, but then the binary needs to be the right architecture.

Comment (vigith) [reply Makefile]: adding `CGO_ENABLED=0` to `go build` fixed the issue.

File: Dockerfile (hunk 1 above)
Comment (vigith) [reply Dockerfile]: fixed the Makefile. I have also moved `/eventrouter` to `/app/eventrouter`

**Comment by vigith** [comment]
I have made the changes in Makefile and Dockerfile. I have also tested it with the yaml file given in the readme session.

**Comment by johnSchnake** [comment]
👍 Seems good to me. Thanks!

**Pull request merged**